examples/basics/emag_basics.py

```diff
@@ -3,220 +3,249 @@
 
 import ansys.motorcad.core as pymotorcad
 
-if "QT_API" in os.environ:
-    os.environ["QT_API"] = "pyqt"
-
 # %%
 # Specify working directory
 # ~~~~~~~~~~~~~~~~~~~~~~~~~
-
-working_folder = os.getcwd()
-
-if os.path.isdir(working_folder) is False:
-    print("Working folder does not exist. Choose a folder that exists and try again.")
-    print(working_folder)
-    exit()
+# Save the file to a temporary folder
+working_folder = os.path.join(tempfile.gettempdir(), "basic_examples")
+try:
+    shutil.rmtree(working_folder)
+except:
+    pass
+os.mkdir(working_folder)
+mot_name = "EMagnetic"
 
 # %%
 # Launch Motor-CAD
 # ~~~~~~~~~~~~~~~~
-
+# Connect to Motor-CAD and open a new Motor-CAD instance. To keep a new Motor-CAD instance open
+# after executing the script, use the ``MotorCAD(keep_instance_open=True)`` option when opening the
+# new instance.
 print("Starting initialization.")
-mcad = pymotorcad.MotorCAD()
+mc = pymotorcad.MotorCAD()
 
 # %%
 # Disable popup messages
 # ~~~~~~~~~~~~~~~~~~~~~~
-
-mcad.set_variable("MessageDisplayState", 2)
-print("Initialization completed.")
-print("Running simulation.")
-
-# %%
-# Create analysis
-# ---------------
-# Creating the analysis consists of showing the magnetic context, displaying
-# the **Scripting** tab, setting the geometry and parameters, and saving
-# the file.
+# Disable all popup messages from Motor-CAD by setting the ``MessageDisplayState`` parameter to 2.
 #
-# Show the magnetic context.
-mcad.show_magnetic_context()
+# .. note::
+#    This will also disable crucial popups, for example prompts to save data or overwrite data, or
+#    dialogs used to reconcile differences in material data between the database and MOT file. In
+#    each case the default action will be taken. This setting will persist until Motor-CAD is
+#    closed.
+mc.set_variable("MessageDisplayState", 2)
 
 # %%
-# Display the **Scripting** tab.
-mcad.display_screen("Scripting")
+# Set Motor-CAD to EMag
+# ~~~~~~~~~~~~~~~~~~~~~
+# Show the E-Magnetic context in Motor-CAD and ensure that the BPM
+# Motor Type is set.
+mc.show_magnetic_context()
+mc.set_variable("Motor_Type", 0)
 
 # %%
-# Set the geometry.
-mcad.set_variable("Slot_Number", 24)
-mcad.set_variable("Tooth_Width", 6)
-mcad.set_variable("Magnet_Thickness", 4.5)
+# Set up analysis
+# ---------------
+# Setting up the analysis consists of setting the geometry, winding and material parameters, setting
+# the calculation options and operating point, and saving the file.
+#
+# Geometry setup
+# ~~~~~~~~~~~~~~
+# Display the **Scripting** tab and set the ``Slot_Number``, ``Tooth_Width`` and
+# ``Magnet_Thickness`` geometry parameters.
+#
+# .. note::
+#    To set parameter values via automation with the GUI visible, a different tab must be displayed.
+#    The **Scripting** tab is chosen because no parameters on this tab will be changed in this
+#    script. It is best practice to display the **Scripting** tab when changing input parameters by
```

Comment:
```suggestion
#    To set parameter values by automation with the GUI visible, a different tab must be displayed.
#    The **Scripting** tab is chosen because no parameters on this tab are changed in this
#    script. It is best practice to display the **Scripting** tab when changing input parameters by
```
Use present tense. Style guide says to avoid using "via" as a preposition.